libs/sdk-ui-pivot/src/impl/resizing/columnSizing.ts

```diff
@@ -872,22 +892,26 @@ function getTableFonts(containerRef: HTMLDivElement): {
 /**
  * Ag-Grid API set desired column sizes (it *mutates* pivot table columns data).
  */
-export async function autoresizeAllColumns(columnApi: ColumnApi | null, autoResizedColumns: IResizedColumns) {
+export async function autoresizeAllColumns(
+    columnApi: ColumnApi | null,
+    autoResizedColumns: IResizedColumns,
+): Promise<void> {
     if (columnApi) {
         const columns = columnApi.getPrimaryColumns();
 
         // Resizing large number of columns is performance-intensive, so split the processing
         // to async chunks to not block main thread until the whole resizing is completed.
         const chunks = chunk(columns, COLUMN_RESIZE_CHUNK_SIZE);
         for (const ch of chunks) {
-            await new Promise((resolve) => {
+            await new Promise<void>((resolve) => {
                 setTimeout(() => {
                     ch.forEach((column: Column) => {
                         const columnDef = column.getColDef();
                         const colId = agColId(columnDef);
                         const autoResizedColumn = autoResizedColumns[colId];
 
                         if (colId && autoResizedColumn && autoResizedColumn.width) {
+                            // TODO After upgrade of aggrid to 23.1 or newer consider usage of columnApi.setColumnWidths with improved performance
```

Comment:
Do we have a task for it? Can we mention the ticket number in the TODO?

Reply:
Created the ticket and mentioned in comment